File: k-th Smallest in BST.java
class Solution {
    ArrayList<Integer> ans;
    // Return the Kth smallest element in the given BST
    public int kthSmallest(Node root, int k) {
        // Write your code here
        ans = new ArrayList<>();
        inorder(root);
        if(ans.size() >= k){
            return ans.get(k-1);
        }
        return -1;
    }
    
    void inorder(Node root){
        if(root==null){
            return;
        }
        inorder(root.left);
        ans.add(root.data);
        inorder(root.right);
    }
}

File: k-th Smallest in BST.py
class Solution:
    def kthSmallest(self, root, k): 
        if root==None or self.count == k:
            return
        self.kthSmallest(root.left, k)
        self.count += 1
        if self.count  == k:
            self.result = root.data
        self.kthSmallest(root.right, k)    
        return self.result    
    count = 0
    result = -1   
